cmd/rtrmon/rtrmon.go

```diff
@@ -414,6 +423,29 @@ func BuildNewVrpMap(log *log.Entry, currentVrps VRPMap, newVrps []prefixfile.VRP
 	return res, inGracePeriod
 }
 
+func UpdateCurrentVrpMap(log *log.Entry, currentVrps VRPMap, now time.Time) (VRPMap, int) {
+	tCurrentUpdate := now.Unix()
+	gracePeriodEnds := tCurrentUpdate - int64(GracePeriod.Seconds())
+	inGracePeriod := 0
+	res := make(VRPMap)
+
+	for key, vrp := range currentVrps {
+		if !vrp.Visible && vrp.LastSeen < gracePeriodEnds {
+			continue
+		}
+
+		updated := *vrp
```

Comment:
Is updated identical to VRP or not? Does it get copied onto the heap here?
(I think it is fine to mutate these objects as long as we have the lock, but do also see why you may not want this)